### PR TITLE
Test example plugins with external builds.

### LINKED ISSUE #1389: Add external plugins to CI tests.
### Description

Add a set of CI tests that build the example plugins externally and runs their tests.

### Motivation and context

These tests will detect more errors like those fixed in #1388. They will also serve to test the example CMake code when used for external plugins.

### PR DESCRIPTION
**Pull request by joaander**
## Description

Build the example plugins external to HOOMD.

## Motivation and context

This ensures that the HOOMD build process installs the header files necessary for plugins to build without error.

Resolves #1389

## How has this been tested?

The CI checks in Github Actions will test these changes.

## Checklist:

- [x] I have reviewed the [**Contributor Guidelines**](https://github.com/glotzerlab/hoomd-blue/blob/trunk-minor/CONTRIBUTING.md).
- [x] I agree with the terms of the [**HOOMD-blue Contributor Agreement**](https://github.com/glotzerlab/hoomd-blue/blob/trunk-minor/ContributorAgreement.md).
- [x] My name is on the list of contributors (`sphinx-doc/credits.rst`) in the pull request source branch.

### CODE REVIEW

**Comment by joaander** [comment]
Confirmation the installed plugin tests are running: https://github.com/glotzerlab/hoomd-blue/runs/8121421002?check_suite_focus=true#step:6:2717

File: example_plugins/pair_plugin/pytest/test_example_pair.py [hunk 1]
```diff
@@ -9,18 +9,11 @@
...
 
 
 # Python implementation of the pair force and energy.
-def harm_force_and_energy(
-        dx: ArrayLike,
-        k: float,
-        sigma: float,
-        r_cut: float,
-        shift: Optional[bool] = False) -> Tuple[NDArray[np.float64], float]:
+def harm_force_and_energy(dx, k, sigma, r_cut, shift=False):
```
Comment (b-butler) [inline example_plugins/pair_plugin/pytest/test_example_pair.py:16]: Why this change?

**Review by b-butler — COMMENTED** [review]
Just one comment on the change to the plugin Python code,

File: example_plugins/pair_plugin/pytest/test_example_pair.py (hunk 1 above)
Comment (joaander) [reply example_plugins/pair_plugin/pytest/test_example_pair.py:16]: The above `from numpy.typing import NDArray, ArrayLike` fails in the Python 3.6 test image: https://github.com/glotzerlab/hoomd-blue/runs/8122103142?check_suite_focus=true#step:6:28

This error was not caught in #1382 because the Python 3.6 tests were not run on that pull request. I changed the build configurations to always include the older python versions in c256cb9 to catch these types of errors sooner.

Comment (b-butler) [reply example_plugins/pair_plugin/pytest/test_example_pair.py:16]: Ah okay. That makes sense. Also do we document our supported Python policy (I know the requirements is in the building from source docs)? NumPy stopped supporting 3.6 since Jun 23, 2020, so we have to grab pretty old packages to use 3.6.

**Review by b-butler — APPROVED** [review]

File: example_plugins/pair_plugin/pytest/test_example_pair.py (hunk 1 above)
Comment (joaander) [reply example_plugins/pair_plugin/pytest/test_example_pair.py:16]: In [ARCHITECTURE.md](https://github.com/glotzerlab/hoomd-blue/blob/trunk-patch/ARCHITECTURE.md#hardware-and-software-support) I state that we support 
> Compilers and software dependencies available on the oldest maintained Ubuntu LTS release. 
That is currently Ubuntu 18.04 which installs Python 3.6 by default.

Following this documentation, we can require Python 3.7 and support Ubuntu 20.04+ starting in April 2023.
